src/test/java/itest/ShouldAcceptMultipartWithBoundaryIT.java

```diff
@@ -0,0 +1,172 @@
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import io.cryostat.net.web.http.HttpMimeType;
+
+import io.netty.handler.timeout.TimeoutException;
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.handler.HttpException;
+import itest.bases.StandardSelfTest;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class ShouldAcceptMultipartWithBoundaryIT extends StandardSelfTest {
+    static final String TEST_RULE_NAME = "Test_Rule";
+
+    static final Map<String, String> NULL_RESULT = new HashMap<>();
+
+    static {
+        NULL_RESULT.put("result", null);
+    }
+
+    // @BeforeAll
+    // static void setup() throws Exception {}
+
+    @AfterEach
+    void cleanup() throws Exception {
+        CompletableFuture<JsonObject> deleteResponse = new CompletableFuture<>();
+        webClient
+                .delete(String.format("/api/v2/rules/%s", TEST_RULE_NAME))
+                .send(
+                        ar -> {
+                            if (ar.succeeded()) {
+                                deleteResponse.complete(ar.result().bodyAsJsonObject());
+                            } else {
+                                deleteResponse.completeExceptionally(ar.cause());
+                            }
+                        });
+
+        JsonObject expectedDeleteResponse =
+                new JsonObject(
+                        Map.of(
+                                "meta",
+                                Map.of("type", HttpMimeType.JSON.mime(), "status", "OK"),
+                                "data",
+                                NULL_RESULT));
+
+        try {
+            JsonObject deleteResult = deleteResponse.get(5, TimeUnit.SECONDS);
+            MatcherAssert.assertThat(deleteResult, Matchers.equalTo(expectedDeleteResponse));
+        } catch (TimeoutException e) {
+            System.out.println("Deletion timed out. Reason: " + e.getMessage());
+            throw new RuntimeException("Deletion did not complete within the expected time.", e);
+        } catch (ExecutionException e) {
+            if (e.getCause() instanceof HttpException) {
+                HttpException httpException = (HttpException) e.getCause();
+                MatcherAssert.assertThat(httpException.getStatusCode(), Matchers.equalTo(400));
+                MatcherAssert.assertThat(
+                        httpException.getMessage(), Matchers.equalTo("Bad Request"));
+            } else {
+                System.out.println(
+                        "Deletion execution failed. Reason: " + e.getCause().getMessage());
+                throw e;
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
```

Comment:
Why does the current thread need to be interrupted if we are catching an exception due to it just being interrupted by another thread?